tests/transport/test_context.cpp

```diff
@@ -0,0 +1,196 @@
+/* 
+ * GridTools
+ * 
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ * 
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * 
+ */
+#include <ghex/threads/std_thread/primitives.hpp>
+#include <ghex/threads/atomic/primitives.hpp>
+#include <iostream>
+#include <iomanip>
+
+#include <gtest/gtest.h>
+
+#ifdef GHEX_TEST_USE_UCX
+#include <ghex/transport_layer/ucx/context.hpp>
+using transport = gridtools::ghex::tl::ucx_tag;
+#else
+#include <ghex/transport_layer/mpi/context.hpp>
+using transport = gridtools::ghex::tl::mpi_tag;
+#endif
+
+using threading_std = gridtools::ghex::threads::std_thread::primitives;
+using threading_atc = gridtools::ghex::threads::atomic::primitives;
+using threading = threading_std;
+//using threading = threading_atc;
+using context_type = gridtools::ghex::tl::context<transport, threading>;
+using comm_type = typename context_type::communicator_type;
+using msg_type = typename comm_type::message_type;
```

Comment:
It seems these types are defined here, but then every test instantiate it's own context type. Would it make more sense to make every test to instantiate their own types?

Reply:
I changed the test: is this what you meant?